sky/provision/aws/instance.py

```diff
@@ -27,6 +28,40 @@ def _filter_instances(ec2, filters: List[Dict[str, Any]],
     return instances
 
 
+def query_instances(
+    cluster_name: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+) -> Dict[str, Optional[status_lib.ClusterStatus]]:
+    assert provider_config is not None, (cluster_name, provider_config)
+    region = provider_config['region']
+    ec2 = aws.resource(
```

Comment:
Nit: seems like we can factor out the creation of `ec2` and/or a basic `filters` with the cluster name? used by all funcs so far and probably for future funcs.

Reply:
good idea. I created two helper functions